File: LeetCode/88-merge.cpp
//从前往后逐一比较两个向量末项的大小，较大者填入nums1向量中
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    int p1 = m - 1;
    int p2 = n - 1;
    int tail = m + n - 1;
    int cur;
    while(p2 >= 0)
    {
        if (p1 == -1)
        {
            cur = nums2[p2--];
        }
        //p2=-1，即nums2的数组提前耗尽，而本身nums1的数都已经就位，因此可以忽略此步的判断
        // else if (p2 == -1)
        // {
        //     cur = nums1[p1--];
        // }
        else if (nums1[p1] > nums2[p2])
        {
            cur = nums1[p1--];
        }
        else
        {
            cur = nums2[p2--];
        }
        nums1[tail--] = cur;
    }
    }
};

File: 面试高频leetcode/双指针/88_merge.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p = m + n - 1
        l1 = m - 1
        l2 = n - 1

        while l1 >= 0 or l2 >= 0:
            if l1 == -1:
                nums1[p] = nums2[l2]
                l2 -= 1
            elif l2 == -1:
                nums1[p] = nums1[l1]
                l1 -= 1
            elif nums1[l1] > nums2[l2]:
                nums1[p] = nums1[l1]
                l1 -= 1
            else:
                nums1[p] = nums2[l2]
                l2 -= 1
            p -= 1
        

            
